### PR TITLE
BPM Input Changes

### PR DESCRIPTION
**Pull request by alexthescott**
- Cache BPM via LocalStoraage
- Moved BPM Interface to models.ts
- Use invalid: state to add guiding error messages for both Max Min inputs
- Disable Yes button if BPM Input is in an invalid state

https://user-images.githubusercontent.com/43523995/209745604-8bf48dfe-2703-4072-a73f-6eed17696b12.mov

### CODE REVIEW

File: src/components/BpmInput.tsx [hunk 1]
```diff
@@ -1,44 +1,47 @@
...
-    min: number
-    max: number
-}
+import { cacheCurrentBpm, getCurrentBPM } from '../utils/cache'
+import { BPM } from '../services/spotify/models'
 
 interface Props extends PropsWithChildren {
-    onChange?(bpm: Bpm): void 
+    onChange?(bpm: BPM): void 
     min?: number
```
Comment (Ykrej) [inline src/components/BpmInput.tsx]: Maybe remove the min/max props if we aren't gonna use them.

File: src/components/BpmInput.tsx [hunk 2]
```diff
@@ -1,44 +1,47 @@
...
     max?: number
 }
-function BpmInput({ onChange, min=60, max=260 }: Props) {
+function BpmInput({ onChange }: Props) {
+    const bpm = getCurrentBPM()
```
Comment (Ykrej) [inline src/components/BpmInput.tsx]: Seems odd to be loading from localStorage every render

File: src/services/spotify/models.ts [hunk 3]
```diff
@@ -62,4 +62,9 @@ interface PlaylistItem {
     track: Track
 }
 
-export type { User, Playlist, Track, TrackFeatures, PlaylistItem }
+interface BPM {
```
Comment (Ykrej) [inline src/services/spotify/models.ts]: This file was meant for spotify api models. This would be better in bpminput or cache

**Comment by Ykrej** [comment]
@alexthescott Looks good. The useEffect for calling onChange needs to be removed because useEffect adds a delay to the onChange call. onChange should probably be called in the input handler. Feel free to skip this change though since it isnt related to your changes here. 
https://github.com/alexthescott/Sprintify/blob/07217a0b17d8b255dfec38a2ef4572537f7a3377/src/components/BpmInput.tsx#L49

EDIT: For context "Sometimes, we want to run some additional code **after** React has updated the DOM." from https://reactjs.org/docs/hooks-effect.html#effects-without-cleanup

**Review by Ykrej — APPROVED** [review]

**Comment by alexthescott** [comment]
Beautiful I'll make another issue for myself to clean the useEffect() hook

**Pull request merged**